pkg/dataobj/internal/dataset/page_test.go

```diff
@@ -14,99 +18,113 @@ import (
 )
 
 func Benchmark_page_Decode(b *testing.B) {
-	in := []string{
-		"hello, world!",
-		"",
-		"this is a test of the emergency broadcast system",
-		"this is only a test",
-		"if this were a real emergency, you would be instructed to panic",
-		"but it's not, so don't",
-		"",
-		"this concludes the test",
-		"thank you for your cooperation",
-		"goodbye",
-	}
+	b.Run("workers=1", func(b *testing.B) { benchmarkPageDecodeParallel(b, 1) })
+	b.Run("workers=2", func(b *testing.B) { benchmarkPageDecodeParallel(b, 2) })
+	b.Run("workers=5", func(b *testing.B) { benchmarkPageDecodeParallel(b, 5) })
+	b.Run("workers=10", func(b *testing.B) { benchmarkPageDecodeParallel(b, 10) })
+}
 
-	opts := BuilderOptions{
-		PageSizeHint: 1 << 30, // 1GiB
-		Value:        datasetmd.VALUE_TYPE_STRING,
-		Compression:  datasetmd.COMPRESSION_TYPE_ZSTD,
-		Encoding:     datasetmd.ENCODING_TYPE_PLAIN,
-	}
-	builder, err := newPageBuilder(opts)
-	require.NoError(b, err)
+func benchmarkPageDecodeParallel(b *testing.B, workers int) {
+	b.StopTimer()
 
-	for i := range 1_000_000 {
-		s := in[i%len(in)]
-		require.True(b, builder.Append(StringValue(s)))
-	}
+	page := logsTestPage(b)
 
-	page, err := builder.Flush()
-	require.NoError(b, err)
+	// Warm up one page reader per worker to avoid benchmarking the cost of
+	// initializing zstd.
+	{
+		var wg sync.WaitGroup
+
+		readers := make([]io.ReadCloser, workers)
+
+		for worker := range workers {
+			wg.Add(1)
+
+			go func(worker int) {
+				defer wg.Done()
 
-	b.ReportAllocs()
-	b.ResetTimer()
+				_, values, err := page.reader(datasetmd.COMPRESSION_TYPE_ZSTD)
+				if err != nil {
+					b.Error(err)
+				} else if _, err := io.Copy(io.Discard, values); err != nil {
+					b.Error(err)
+				}
 
-	for range b.N {
-		_, values, err := page.reader(datasetmd.COMPRESSION_TYPE_ZSTD)
-		if err != nil {
-			b.Fatal()
+				readers[worker] = values
+			}(worker)
 		}
-		if _, err := io.Copy(io.Discard, values); err != nil {
-			b.Fatal(err)
-		} else if err := values.Close(); err != nil {
-			b.Fatal(err)
+
+		wg.Wait()
+
+		// Close all the readers at once; closing a reader releases it back to the
+		// pool so we only do this after the workers are done to guarantee that
+		// there is exactly one zstd reader in the pool per worker.
+		for worker := range workers {
+			require.NoError(b, readers[worker].Close())
 		}
 	}
 
-}
+	b.StartTimer()
 
-func Benchmark_pageBuilder_WriteRead(b *testing.B) {
```

Comment:
This benchmark didn't seem to be doing anything that Benchmark_page_Decode wasn't already doing, so I removed it